File: backend/src/main/java/com/app/vending/iot/entity/Order.java
package com.app.vending.iot.entity;

import com.app.vending.iot.common.enums.OrderStatus;
import com.app.vending.iot.dto.ProductOrder;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "orders")
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Order {

    @Id
    String id;
    String machineId;
    @Builder.Default
    List<ProductOrder> products = new ArrayList<>();
    LocalDateTime date;
    Double total;
    OrderStatus status;

}

File: frontend/win/vending/vending/model/Order.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace vending.model
{
    public class Order
    {
        [JsonPropertyName("id")]
        public string Id { get; set; }

        [JsonPropertyName("machineId")]
        public string MachineId { get; set; }

        [JsonPropertyName("products")]
        public List<Product> Products { get; set; }

        [JsonPropertyName("date")]
        public DateTime Date { get; set; }

        [JsonPropertyName("total")]
        public double Total { get; set; }

        [JsonPropertyName("status")]
        public string Status { get; set; }

        [JsonPropertyName("datePay")]
        public DateTime? DatePay { get; set; }
    }
}